ingestion/src/metadata/ingestion/source/sql_source.py

```diff
@@ -102,6 +106,95 @@ def _get_table_description(schema: str, table: str, inspector: Inspector) -> str
     return description
 
 
+def get_dbt_local(config):
+    if config.dbtConfig.dbtCatalogFilePath is not None:
+        with open(
+            config.dbtConfig.dbtCatalogFilePath, "r", encoding="utf-8"
+        ) as catalog:
+            dbt_catalog = catalog.read()
+    if config.dbtConfig.dbtManifestFilePath is not None:
+        with open(
+            config.dbtConfig.dbtManifestFilePath, "r", encoding="utf-8"
+        ) as manifest:
+            dbt_manifest = manifest.read()
+    return json.loads(dbt_catalog), json.loads(dbt_manifest)
+
+
+def get_dbt_http(config):
+    import urllib.request
+
+    catalog_file = urllib.request.urlopen(config.dbtConfig.dbtCatalogFilePath)
```

Comment:
We could wrap the `urlopen` with a try/catch and have a explicit exception if we cannot read the resources

Reply:
okay